### PR TITLE
Adding hybrid implementation of 'cleanReSyncGhosts' API

### PR DESCRIPTION
**Pull request by bhariharan**
```JS``` and ```iOS``` coming up.

### CODE REVIEW

File: libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/plugin/SmartSyncPlugin.java [hunk 1]
```diff
@@ -170,29 +192,50 @@ private void getSyncStatus(JSONArray args, CallbackContext callbackContext) thro
     }
 
     /**
-     * Native implementatino of reSync
+     * Native implementatino of reSync.
```
Comment (kchitalia) [inline libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/plugin/SmartSyncPlugin.java]: Typo here :P. Not an issue though.

**Comment by bhariharan** [comment]
@kchitalia Fixed [here](https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/1107/commits/422b4409201abedb700f05c497f12e299ddd2b8f).

**Comment by wmathurin** [comment]
👍

**Comment by kchitalia** [comment]
:+1:

**Pull request merged**